test/qa/lib/meshnet.py

```diff
@@ -222,4 +276,159 @@ def deny_meshnet_invite(ssh_client: ssh.Ssh):
+            if (not local) and (lan_drop_rule_index > routing_enabled_rule_index):
+                return False, f"LAN drop rule for subnet {lan} added after routing\nrules: {rules}"
+            continue
+
+        # If routing is not enabled, but lan is enabled, there should be one rule for each local network for the peer.
+        # They should be located above the LAN block rules.
+        if not local:
+            continue
+
+        lan_allow_rule = f"-A FORWARD -s {peer_ip}/32 -d {lan} -m comment --comment nordvpn-exitnode-transient -j ACCEPT"
+        lan_allow_rule_index = rules.find(lan_allow_rule)
+
+        if lan_allow_rule not in rules:
+            return False, f"LAN allow rule for subnet {lan} not found\nrules:\n{rules}"
+
+        if lan_allow_rule_index > lan_drop_rule_index:
+            return False, f"LAN allow rule is added after LAN drop rule\nrules:\n{rules}"
+
+    return True, ""
+
+
+def set_permissions(peer: str, routing: bool, local: bool, incoming: bool, fileshare: bool):
+    def bool_to_permission(permission: bool) -> str:
+        if permission:
+            return "allow"
+        return "deny"
+
+    # ignore any failures that might occur when permissions are already configured to the desired value
+    sh.nordvpn.mesh.peer.routing(bool_to_permission(routing), peer, _ok_code=(0, 1))
+    sh.nordvpn.mesh.peer.local(bool_to_permission(local), peer, _ok_code=(0, 1))
+    sh.nordvpn.mesh.peer.incoming(bool_to_permission(incoming), peer, _ok_code=(0, 1))
+    sh.nordvpn.mesh.peer.fileshare(bool_to_permission(fileshare), peer, _ok_code=(0, 1))
+
+
+def get_clean_peer_list(peer_list: str):
+    output = strip_colors.sub('', str(peer_list))
+    output = "This " + output.split("This", 1)[-1].strip()
+    return output
+
+
+def parse_peer_list(output: str):
+    """ Converts output/meshnet peer list string to PeerList object. """
+
+    def remove_text_before_and_keyword(input_string, keyword):
+        index = input_string.find(keyword)
+
+        if index != -1:
+            return input_string[index + len(keyword):]
+        else:
+            return input_string
+
+    peer_list = get_clean_peer_list(output)
+    peer_list_object = PeerList()
+
+    this_device = peer_list.split("\n\n")[0].replace("This device:\n", "")
+    peer_list_object.set_this_device(this_device)
+
+    internal_peers = remove_text_before_and_keyword(peer_list, "Local Peers:\n").split("\n\n\n")[0]
+    if "[no peers]" not in internal_peers:
+        internal_peer_list = internal_peers.split("\n\n")
+
+        for peer_data in internal_peer_list:
+            peer_list_object.add_internal_peer(peer_data)
+
+    external_peers = remove_text_before_and_keyword(peer_list, "External Peers:\n")
+    if "[no peers]" not in external_peers:
+        external_peer_list = external_peers.split("\n\n")
+
+        for peer_data in external_peer_list:
+            peer_list_object.add_external_peer(peer_data)
+
+    return peer_list_object
```

Comment:
Similarly, you can make a `PeerList.from_str()` classmethod for conctructing a `PeerList` from meshnet output peer list.